### PR TITLE
Avoid deadlock on close

### PR DESCRIPTION
**Pull request by harold-toledo**
If streaming cluster is not available or reachable when calling Close() or Dispose(), previous implementation was waiting indefinitely for a response to a request to close made to the streaming cluster. A unit test was added to verify this:

- TestStreamingServerGone

### CODE REVIEW

**Comment by harold-toledo** [comment]
I based this PR on PR #71, to build on top of that one, that is why it has all the commits already present in PR #71.

**Comment by harold-toledo** [comment]
But if you guys prefer a fix on the actual code, I can also do that and create a new PR.

**Comment by ColinSullivan1** [comment]
Thanks!  If it's a small change, I'd prefer a new PR, and then merge #71 separately.

File: STAN.CLIENT/Options.cs [hunk 1]
```diff
@@ -87,12 +88,30 @@ public int ConnectTimeout
...
+        public int CloseTimeout
+        {
+            get
+            {
+                return closeTimeout;
+            }
+            set
+            {
+                if (value <= 0)
+                    throw new ArgumentOutOfRangeException(nameof(value), value, "ConnectTimeout must be greater than zero.");
```
Comment (watfordgnf) [inline STAN.CLIENT/Options.cs]:
```suggestion
                    throw new ArgumentOutOfRangeException(nameof(value), value, "CloseTimeout must be greater than zero.");
```

**Comment by ColinSullivan1** [comment]
I was going to reach out to you after this last release.  I realize the code base has changed, but would you still be interesting in pursuing this fix?

**Comment by harold-toledo** [comment]
I would try to propose my original changes again, starting with some of the stylistic changes as a base if you guys find that ok. After that I would try to introduce some changes incrementally, if that is possible. Thanks for reaching out.

**Comment by ColinSullivan1** [comment]
I think that'd be great - incremental PRs are often better - much easier to review and good regarding separation of concerns.  Also if you would like to propose big changes, certainly feel free to open an issue for discussion ahead of time or bring it up on slack.

**Pull request closed without merging**